File: gaia-sdk-javascript/src/graphql/response/type/CreatedEdgeImpulse.ts

import {Edge} from "./Edge";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a create edge impulse
*/
export interface CreatedEdgeImpulse {
    id?:Uuid, 
    /**
    * the edge instance
    */
    data?:Edge
}

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/CreatedEdgeImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result of a create edge impulse
 */
class CreatedEdgeImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    /**
     * the edge instance
     */
    fun data(config: Edge.() -> Unit) = 
        add { "data{ " + Edge().apply(config).render(it) + "}"}

}

